include/revng/RestructureCFG/MarkerCallUtils.h

```diff
@@ -0,0 +1,222 @@
+#pragma once
+
+//
+// Copyright rev.ng Labs Srl. See LICENSE.md for details.
+//
+
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/IR/GlobalValue.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/Support/ModRef.h"
+
+#include "revng/Support/FunctionTags.h"
+#include "revng/Support/IRHelpers.h"
+
+std::string ScopeCloserMarkerCallFunctionName = "scope-closer";
+std::string GotoBlockMarkerCallFunctionName = "goto-block";
+
+inline llvm::Function *
+getMarkerCallFunction(FunctionTags::Tag &MarkerFunctionTag, llvm::Module *M) {
+  llvm::Function *MarkerCallFunction = nullptr;
+
+  // We could early break from this loop but we would loose the ability of
+  // asserting that a single marker function is present
+  for (llvm::Function &F : MarkerFunctionTag.functions(M)) {
+    revng_assert(not MarkerCallFunction);
+    MarkerCallFunction = &F;
+  }
+
+  revng_assert(MarkerCallFunction);
+  return MarkerCallFunction;
+}
+
+inline std::string getMarkerCallFunctionName(llvm::Function *F) {
+  return F->getName().str();
+}
+
+/// This is a helper class used to create the markers in the basic blocks needed
+/// to handle virtual edges, such as the scope closer edges or the goto edges.
+/// Its role should be similar somewhat to the `llvm::IRBuilder` role.
+class MarkerCallBuilder {
```

Comment:
This class changed a bit, but I did the rename as you requested here https://github.com/revng/revng/blob/57e9b483c79b868ba12fde41644a3ee0aba57651/include/revng/RestructureCFG/ScopeGraphUtils.h#L43. I didn't understand, from your comment, if you are completely against the use of the `Marker` term in general, there are a couple of occurrences around, because I think that the terms quite correctly describe the meaning of the markers used for the scope closer edges and the goto edge, but let me know your opinion on this